### PR TITLE
Convert solidity-coverage to 0x/sol-coverage

### PR DESCRIPTION
**Pull request by barlock**
**Related Issue**  
Supports #93 

**Related PRs**  
This PR is not dependent on any other PR

**What does this PR do?**  
Replaces `solidity-coverage` which seems to not be maintained anymore to `0x/sol-coverage` which is! It also uses a generally better method of traces (Which could help debug in the future) rather than instrumenting like "normal".

**Description of Changes**  
Lots of shenanigans. I'll comment.

**What gif most accurately describes how I feel towards this PR?**  
gut punch

### CODE REVIEW

File: packages/bootleg-app-contracts/test/MetaCoin.test.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ contract('MetaCoin', accounts => {
   };
 
   beforeEach(async () => {
-    metaCoinInstance = await MetaCoin.deployed();
+    metaCoinInstance = await MetaCoin.new();
```
Comment (barlock) [inline packages/bootleg-app-contracts/test/MetaCoin.test.js]: Done for coverage. But also it's just a better pattern.

File: packages/bootleg-app-contracts/truffle-config.js [hunk 2]
```diff
@@ -1,5 +1 @@
-if (__dirname.includes('coverageEnv')) {
-  module.exports = require('../../../truffle-config');
-} else {
-  module.exports = require('../../truffle-config');
-}
+module.exports = require('bootleg-common/truffle-config')(__dirname);
```
Comment (barlock) [inline packages/bootleg-app-contracts/truffle-config.js]: solidity-coverage copied all the code into a `coverageEnv` folder. sol-coverage doesn't 🎉

File: packages/bootleg-common/bin/bootleg-coverage-ignore [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+const ignorePatterns = ['**/node_modules/**', '**/Migrations.sol'];
+const coveredFiles = Object.keys(coverageData);
+
+if (coveredFiles.length < 1) {
+  console.error("Couldn't find any covered files");
+  process.exit(1);
+}
+
+const ignoredCoverageData = coveredFiles
```
Comment (barlock) [inline packages/bootleg-common/bin/bootleg-coverage-ignore]: sol-coverage just instruments _everything_. I'ma go open an issue on their side for a supported way to do this.

This script just ignores everything in node_modules and `migrations.sol`.

File: packages/bootleg-common/LICENSE [hunk 4]
```diff
@@ -0,0 +1,203 @@
+
```
Comment (breakpointer) [inline packages/bootleg-common/LICENSE]: related?

File: packages/bootleg-common/bin/bootleg-coverage-ignore (hunk 3 above)
Comment (breakpointer) [reply packages/bootleg-common/bin/bootleg-coverage-ignore]: 👍

File: packages/bootleg-common/bin/bootleg-setup-ganache [hunk 5]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+
+cleanup_ganache() {
+  # Kill the ganache instance that we started (if we started one and if it's still running).
+  if [ -n "$ganache_pid" ] && ps -p $ganache_pid > /dev/null; then
+    echo "Stopping Ganache"
+    kill -9 $ganache_pid
```
Comment (breakpointer) [inline packages/bootleg-common/bin/bootleg-setup-ganache]: hmm... pretty aggressive. Is `-9` necessary?

File: packages/bootleg-common/bin/bootleg-truffle-test [hunk 6]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+# Exit script as soon as a command fails.
+set -o errexit
```
Comment (breakpointer) [inline packages/bootleg-common/bin/bootleg-truffle-test]: I've never seen this before. Is it similar to `set -e`?

Comment (barlock) [reply packages/bootleg-common/bin/bootleg-truffle-test]: Honestly, I stole the code from https://github.com/0xProject/dev-tools-truffle-example/issues/5#issuecomment-457290796 after seeing that it was clearly better than the crappy bash I'd written originally. THEN, I didn't seak to understand the bits as well 😫 Shame on me. 

https://www.gnu.org/software/bash/manual/bash.html#The-Set-Builtin

Turns out this is equivalent to `set -e` 🤷‍♂️

Comment (breakpointer) [reply packages/bootleg-common/bin/bootleg-truffle-test]: Ah okay, it seems more specific of an exit condition than ‘-e’ but I was not sure. Thanks for investing

File: packages/bootleg-common/bin/bootleg-setup-ganache (hunk 5 above)
Comment (barlock) [reply packages/bootleg-common/bin/bootleg-setup-ganache]: Yes, yes it is.

Comment (breakpointer) [reply packages/bootleg-common/bin/bootleg-setup-ganache]: Aggressive from the perspective of the running process... 😜Shooting it in the head vs asking nicely for it to die. http://man7.org/linux/man-pages/man7/signal.7.html

**Pull request merged**